sources/assets/Xenko.Core.Assets/AssetCompositeHierarchy.cs

```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) Xenko contributors (https://xenko.com) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
+// Copyright (c) Xenko contributors (https://xenko.com) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
```

Comment:
I'll have a closer look.

Reply:
@Kryptos-FR On closer inspection some of the files have the BOM and some don't. It probably depends on when each file was last saved.
However, from what testing I've done, the [.editorconfig](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties) is incorrect anyway and should specify `utf-8-bom` to have the desired behaviour. (The current setting actually strips the BOM whenever I save a file.)

Reply:
It looks like there was a bug in VS 2017 that was fixed in 15.3 where `utf-8` was interpreted as `utf-8-bom`, so it would've been working but has stopped working since then. See [VS2017 RC breaks the encoding of my files](https://developercommunity.visualstudio.com/comments/215480/view.html).

Reply:
I think we were stripping the BOM on purpose. I don't remember why, maybe some tools were failing for Unix platforms (Linux, Android). @xen2 if you could have a look at the old repo history or if you remember why we had that...

Reply:
Also editing a file in VS when `charset = utf-8-bom` doesn't seem to update an existing file to add the BOM. I thought it did but now I'm confused as it definitely isn't. Maybe it only adds the BOM to newly created files?

Reply:
I only remember VS had a bug and editorconfig was not working properly (i see lot of switch between bom/non-bom in the project history).
Forgot which way we wanted to ultimately go...
Any recommendations?

Reply:
The majority of the files I looked at in the `sources/assets/Xenko.Core.Assets` do _not_ have a BOM. Also, some people don't like the use of a BOM. Are you sure you want to enforce it? You probably should update all the files to make them consistent then. 🤔

Reply:
I think I read that people prefer no BOM because it's more tooling friendly. Yeah once you decide which way to go I'd update all the source to match ASAP. Could easily write a PowerShell script to do it so I'm happy to submit a PR.